### ISSUE TITLE
`inherent_method_now_returns_unit` conflates same-named methods from different inherent impls

### ISSUE THREAD

**Issue by ssrlive** [obi1kenobi/cargo-semver-checks#1656]
`cargo-semver-checks` can report false-positive `inherent_method_now_returns_unit` lints even when the old and new APIs are identical.

Minimal repro: use this same code for both baseline and current crates.

```rust
pub struct Gc<T>(T);

impl<T> Gc<std::cell::Cell<T>> {
    pub fn set(self, value: T) {
        self.0.set(value);
    }
}

impl<T> Gc<std::sync::OnceLock<T>> {
    pub fn set(self, value: T) -> Result<(), T> {
        self.0.set(value)
    }
}
```

Expected: no semver violation, since both `set` methods are unchanged.

Actual: `inherent_method_now_returns_unit` reports `Gc::set` as changed from returning a value to returning `()`.

The lint appears to match inherent methods by importable owner path plus method name, so it can pair `Gc<OnceLock<T>>::set -> Result<(), T>` from the baseline with `Gc<Cell<T>>::set -> ()` from current. It needs to distinguish the inherent impl target/self type before comparing return types.

Originally encountered in https://github.com/ssrlive/gc-arena/actions/runs/28295846676/job/83835503682
```
A publicly-visible method that used to return a value now returns `()`.
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.48.0/src/lints/inherent_method_now_returns_unit.ron
Failed in:
  gc_arena::Gc::set in /Users/runner/work/gc-arena/gc-arena/src/lock.rs:157
```

**Comment by obi1kenobi** [comment]
Thanks for the bug report. I've extracted a minified repro, and will update the issue title and top post with more information.

If possible in the future, could you please try to use the GitHub issue templates to report issues? It really helps me with triage and maintenance when issues follow the template, supply all required information, and set the correct labels. Otherwise I have to do it all by hand with partial information (e.g. I don't know anything about your repo, Rust version, cargo-semver-checks version, etc.), which consumes time and energy I could spend improving `cargo-semver-checks` directly.
